### PR TITLE
Port run hooks support

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

File: rust_src/.dir-locals.el [hunk 1]
```diff
@@ -0,0 +1,3 @@
+((rust-mode
+  (racer-rust-src-path . "/home/shaleh/\.rustup/toolchains/nightly-2018-02-03-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src")))
+
```
Comment (Wilfred) [inline rust_src/.dir-locals.el]: I don't think you meant to check this in :)

File: rust_src/src/eval.rs [hunk 2]
```diff
@@ -860,4 +861,115 @@ pub fn autoload_do_load(
...
+    run_hook_with_args_internal(args, funcall_nil)
+}
+
+fn funcall_nil(args: &mut [LispObject]) -> LispObject {
+    let mut obj_array: Vec<Lisp_Object> = args.iter().map(|o| o.to_raw()).collect();
+    unsafe { Ffuncall(obj_array.len() as isize, obj_array.as_mut_ptr()) };
+    LispObject::constant_nil()
+}
+
+/* Run the hook HOOK, giving each function no args.  */
```
Comment (Wilfred) [inline rust_src/src/eval.rs]: For consistency, we should probably use `///` comments here.

File: rust_src/.dir-locals.el (hunk 1 above)
Comment (shaleh) [reply rust_src/.dir-locals.el]: Indeed. I thought this was covered with gitignore. Bah!

File: rust_src/src/eval.rs (hunk 2 above)
Comment (shaleh) [reply rust_src/src/eval.rs]: right. Oops.

**Comment by shaleh** [comment]
Yes, there are more hook functions to port.

**Pull request merged**